.github/workflows/Wasm.yml

```diff
@@ -45,33 +50,69 @@ jobs:
-                  AWS_SECRET_ACCESS_KEY: ${{secrets.S3_KEY}}
-                  AWS_DEFAULT_REGION: us-east-1
-                  DUCKDB_EXTENSION_SIGNING_PK: ${{ secrets.DUCKDB_EXTENSION_SIGNING_PK }}
+            - name: Build Wasm module EH
+              if: ${{ matrix.duckdb_wasm_arch == 'eh' }}
+              run: |
+                mkdir -p ./build/wasm_eh
+                emcmake cmake -G "Ninja" -DWASM_LOADABLE_EXTENSIONS=1 -DBUILD_EXTENSIONS_ONLY=1 -Bbuild/wasm_eh -DCMAKE_CXX_FLAGS="-fwasm-exceptions -DWEBDB_FAST_EXCEPTIONS=1 -DDUCKDB_CUSTOM_PLATFORM=wasm_eh" -DDUCKDB_EXPLICIT_PLATFORM=wasm_eh -DLOCAL_EXTENSION_REPO='build/to_be_deployed' -DDUCKDB_EXTENSION_CONFIGS=".github/config/in_tree_extensions.cmake" -DSKIP_EXTENSIONS="httpfs"
+                emmake ninja -j8 -Cbuild/wasm_eh
+
+            - name: Build Wasm module THREADS
+              if: ${{ matrix.duckdb_wasm_arch == 'threads' }}
               run: |
-                  bash ./scripts/extension-upload-wasm.sh ${{ env.DUCKDB_PLATFORM }} ${{ github.event.inputs.duckdb-ref }}
+                mkdir -p ./build/wasm_threads
+                emcmake cmake -G "Ninja" -DWASM_LOADABLE_EXTENSIONS=1 -DBUILD_EXTENSIONS_ONLY=1 -Bbuild/wasm_threads -DCMAKE_CXX_FLAGS="-fwasm-exceptions -DWEBDB_FAST_EXCEPTIONS=1 -DWITH_WASM_THREADS=1 -DWITH_WASM_SIMD=1 -DWITH_WASM_BULK_MEMORY=1 -DDUCKDB_CUSTOM_PLATFORM=wasm_threads" -DDUCKDB_EXPLICIT_PLATFORM=wasm_threads -DLOCAL_EXTENSION_REPO='build/to_be_deployed' -DDUCKDB_EXTENSION_CONFIGS=".github/config/in_tree_extensions.cmake" -DSKIP_EXTENSIONS="httpfs"
+                emmake ninja -j8 -Cbuild/wasm_threads
 
             - name: Upload artifact
               uses: actions/upload-artifact@v3
               with:
-                  name: duckdb-wasm-${{ matrix.duckdb_wasm_arch }}
-                  path: build/${{ env.DUCKDB_PLATFORM }}/built_extensions
+                  name: duckdb_extensions_${{ env.DUCKDB_PLATFORM }}
+                  path: build/to_be_deployed/${{ inputs.duckdb_ref }}/${{ env.DUCKDB_PLATFORM }}
                   retention-days: 1
 
-  trigger_github_pages_build:
-    name: Trigger follow-up work
+  publish:
+    name: Publish extensions
     runs-on: ubuntu-latest
-    needs: build_and_publish
+    needs:
+    - build_wasm
+    strategy:
+      matrix:
+        duckdb_arch: [ 'wasm_mvp', 'wasm_eh', 'wasm_threads' ]
     steps:
-            - name: Move control to duckdb-wasm
+            - uses: actions/checkout@v3
+
+            - uses: actions/download-artifact@v3
+              with:
+                name: duckdb_extensions_${{ matrix.duckdb_arch }}
+                path: build/to_be_deployed/${{ inputs.duckdb_ref }}/${{ matrix.duckdb_arch }}
+
+            - uses: actions/setup-python@v4
+              with:
+                python-version: '3.12'
+
+            - name: Install aws
+              run: |
+                  pip install awscli
+
+            - name: Sign and deploy Wasm extensions
```

Comment:
I find these last two steps slightly confusingly named, it's easy to miss that they both do the same thing, but one of them is the dry run and the other one is the actual deploy, maybe add a `(Dry Run)` to the name of the step here? Otherwise on inspection of ci logs you can't really tell wether the dry run or the real run has been performed?